docker/scvi/scripts/main/artifact_metrics.py

```diff
@@ -0,0 +1,104 @@
+import argparse
+import scanpy
+import numpy as np
+from scib_metrics.benchmark import Benchmarker, BioConservation
+from scib_metrics.nearest_neighbors import NeighborsResults
+import faiss
+from pathlib import Path
+
+parser = argparse.ArgumentParser(description="Run scVI integration")
+parser.add_argument(
+    "--latent-key",
+    dest="latent_key",
+    type=str,
+    default="X_scvi",
+    help="Latent key to save the scvi latent to",
+)
+parser.add_argument(
+    "--batch-key",
+    dest="batch_key",
+    type=str,
+    help="Key in AnnData object for batch information",
+)
+parser.add_argument(
+    "--adata-input", dest="adata_input", type=str, help="AnnData object for a dataset"
+)
+parser.add_argument(
+    "--output-report-dir",
+    dest="scib_report_dir",
+    type=str,
+    help="Output folder to save `scib` report",
+)
+
+# TODO: optional scvi arguments
+
+args = parser.parse_args()
+
+
+# TODO: add these functions to utility/helpers.py
+def faiss_hnsw_nn(X: np.ndarray, k: int):
+    """Gpu HNSW nearest neighbor search using faiss.
+
+    See https://github.com/nmslib/hnswlib/blob/master/ALGO_PARAMS.md
+    for index param details.
+    """
+    X = np.ascontiguousarray(X, dtype=np.float32)
+    res = faiss.StandardGpuResources()
+    M = 32
+    index = faiss.IndexHNSWFlat(X.shape[1], M, faiss.METRIC_L2)
+    gpu_index = faiss.index_cpu_to_gpu(res, 0, index)
+    gpu_index.add(X)
+    distances, indices = gpu_index.search(X, k)
+    del index
+    del gpu_index
+    # distances are squared
+    return NeighborsResults(indices=indices, distances=np.sqrt(distances))
+
+
+def faiss_brute_force_nn(X: np.ndarray, k: int):
+    """Gpu brute force nearest neighbor search using faiss."""
+    X = np.ascontiguousarray(X, dtype=np.float32)
+    res = faiss.StandardGpuResources()
+    index = faiss.IndexFlatL2(X.shape[1])
+    gpu_index = faiss.index_cpu_to_gpu(res, 0, index)
+    gpu_index.add(X)
+    distances, indices = gpu_index.search(X, k)
+    del index
+    del gpu_index
+    # distances are squared
+    return NeighborsResults(indices=indices, distances=np.sqrt(distances))
+
+
+adata = scanpy.read_h5ad(args.adata_input)  # type: ignore
+
+# these should be there...
+if "X_pca" not in adata.obsm:
+    scanpy.pp.pca(adata, n_comps=30)
+
+if "X_pca_harmony" not in adata.obsm:
+    scanpy.external.pp.harmony_integrate(adata, "sample")
+
+
+adata.obsm["Unintegrated"] = adata.obsm["X_pca"]
+
+biocons = BioConservation(isolated_labels=False)
+
+bm = Benchmarker(
+    adata,
+    batch_key="sample",
```

Comment:
@ergonyc Is this supposed to hardcoded as sample? There's an unused argument `--batch-key` and `"batch_id"` has been used, see [process.py](https://github.com/ASAP-CRN/harmonized-wf-dev/blob/8c9c589523e5543af06222c662a7deecb02e5996/docker/scvi/scripts/main/process.py#L78-L83)

Reply:
Yes we should remove the hard code. However for our current data the results are poor if sample is chosen (hence the hard code left over from testing).

Reply:
@ergonyc Thanks, so should I use `sample` or `batch_id` as `--batch-key`?

Reply:
Use "batch_id" as --batch_key